lib/AST/ASTContext.cpp

```diff
@@ -1069,7 +1069,6 @@ ASTContext::getBuiltinInitDecl(NominalTypeDecl *decl,
   if (!witness) {
     assert(false && "Missing required witness");
     witness = ConcreteDeclRef();
```

Comment:
I think this entire block can be folded to `assert(witness && "Missing required witness")`.

Reply:
How so?

Reply:
The assert is always hit.